src/target/rp.c

```diff
@@ -382,34 +407,11 @@ bool rp_probe(target *t)
 		return false;
 	}
  	t->target_storage = (void*)priv_storage;
-	uint32_t bootsec[16];
-	target_mem_read( t, bootsec, XIP_FLASH_START, sizeof( bootsec));
-	int i;
-	for (i = 0; i < 16; i++)
-		if (bootsec[i])
-			break;
-	uint32_t size = 8 * 1024 *1024;
-	if (i == 16) {
-		DEBUG_WARN("Use default size\n");
-	} else {
-		/* Find out size of connected SPI Flash
-		 *
-		 * Flash needs valid content to be mapped
-		 * Low flash is mirrored when flash size is exceeded
-		 */
-		while (size) {
-			uint32_t mirrorsec[16];
-			target_mem_read(t, mirrorsec, XIP_FLASH_START + size,
-							sizeof( bootsec));
-			if (memcmp(bootsec, mirrorsec, sizeof( bootsec)))
-				break;
-			size >>= 1;
-		}
-	}
-	rp_add_flash(t, XIP_FLASH_START, size << 1);
+
+	rp_add_flash(t, XIP_FLASH_START, MAX_FLASH);
```

Comment:
I agree completely with you, it would be a very nice thing to have. Interestingly, Raspberry Pi implemented a [function in the bootrom](https://github.com/raspberrypi/pico-bootrom/blob/master/bootrom/program_flash_generic.c#L372) to check the the flash size by reading the SFDP table (with a back-up of checking JEDEC ID), but this was never included in the bootrom function lookup table...

Maybe this issue, plus the ambiguity of the SPI erase commands gives enough weight to the argument to implement reading the SFDP table when probing the device? This would of course interrupt the program execution and make it impossible to connect to a running target however...

Reply:
Agreed, where an SFDP table exists that would be ideal.. Failing that, fall back on some kind of detection (the JEDEC ID bytes are: the MFR ID, the MFR Flash type code, and the capacity (log2); so using the capacity byte works for all chips that you can reasonably expect to work with the RP2040)

Reply:
we're of the mind to make this a non-blocker and merge as this code will work for most devices and further refinement can be done later by trying this hybrid SFDP-then-JEDEC-ID approach to better solve Flash sizing and erase command issues.

With that in mind, resolving this conversation and approving for merge.

Reply:
Awesome 👍 I can experiment with reading the SFDP table and JEDEC ID and will open a new issue as and when I make progress